src/V3Tristate.cpp

```diff
@@ -636,6 +639,129 @@ class TristateVisitor final : public TristateBaseVisitor {
         nodep->addStmtp(assp);
     }
 
+    void addToAssignmentList(AstAssignW* nodep) {
+        if (AstVarRef* const varRefp = VN_CAST(nodep->lhsp(), VarRef)) {
+            if (varRefp->varp()->isNet()) {
+                m_assigns[varRefp->varp()].push_back(nodep);
+            } else if (nodep->strengthSpecp()) {
+                if (!varRefp->varp()->isNet())
+                    nodep->v3warn(E_UNSUPPORTED, "Unsupported: Signal strengths are unsupported "
+                                                 "on the following variable type: "
+                                                     << varRefp->varp()->varType());
+
+                nodep->strengthSpecp()->unlinkFrBack()->deleteTree();
+            }
+        } else if (nodep->strengthSpecp()) {
+            nodep->v3warn(E_UNSUPPORTED,
+                          "Unsupported: Assignments with signal strength with LHS of type: "
+                              << nodep->lhsp()->prettyTypeName());
+        }
+    }
+
+    uint8_t getStrength(AstAssignW* const nodep, bool value) {
+        if (AstStrengthSpec* const strengthSpec = nodep->strengthSpecp()) {
+            if (value)
+                return strengthSpec->strength1();
+            else
+                return strengthSpec->strength0();
+        } else {
+            return 6;  // default strength in strong (6)
+        }
+    }
+
+    bool assignmentOfValueOnAllBits(AstAssignW* const nodep, bool value) {
+        if (AstConst* const constp = VN_CAST(nodep->rhsp(), Const)) {
+            const V3Number num = constp->num();
+            if (value)
+                return num.isEqAllOnes();
+            else
+                return num.isEqZero();
+        }
+        return false;
+    }
+
+    AstAssignW* getStrongestAssignmentOfValue(const Assigns& assigns, bool value) {
+        auto maxIt = std::max_element(
+            assigns.begin(), assigns.end(), [&](AstAssignW* ap, AstAssignW* bp) {
+                bool valuesOnRhsA = assignmentOfValueOnAllBits(ap, value);
+                bool valuesOnRhsB = assignmentOfValueOnAllBits(bp, value);
+                if (!valuesOnRhsA && !valuesOnRhsB) return false;
+                if (!valuesOnRhsA) return true;
+                if (!valuesOnRhsB) return false;
+                return getStrength(ap, value) < getStrength(bp, value);
+            });
+        // If all assignment don't have const with all bits equal to value on the RHS,
+        // std::max_element will return one of them anyway, so we need to check before returning
+        if (assignmentOfValueOnAllBits(*maxIt, value))
+            return *maxIt;
+        else
+            return nullptr;
```

Comment:
```suggestion
        // If not all assignments have const with all bits equal to value on the RHS,
        // std::max_element will return one of them anyway, so we need to check before returning
        if (assignmentOfValueOnAllBits(*maxIt, value)) {
            return *maxIt;
        } else {
            return nullptr;
        }
```
Good place for a `? :`, but I think it doesn't work with `nullptr`.

Reply:
I will check, thanks.

Reply:
It works.

Reply:
It works.